### PR TITLE
Add RPC endpoints for leave and connect

### LINKED ISSUE #508: Add rpc endpoints for `connect` and `leave` of token networks
## Problem Definition
Channel Provisioning needs to be accessible from the rpc.

## Solution

Add endpoints for `leave` and `connect`

#### Endpoint for `connect()`:
PUT /api/v1/connection/<:token_address>

with a json payload of the provisioning details which contain:

```
{
    "funds": 1337,
    "initial_channel_target": 3,
    "joinable_funds_target": 0.4,
}
```

#### Endpoint for `leave()`:
DELETE /api/v1/connection/<:token_address>

with a json payload of the closing details which contain:

```
{
    "wait_for_settle": true,
    "timeout": 30,
}
```

### PR DESCRIPTION
**Pull request by czepluch**
This PR adds RPC endpoints for `leave` and `connect`. 

It also adds tests for these endpoints and fixed #508

### CODE REVIEW

File: raiden/tests/api/test_api.py [hunk 1]
```diff
@@ -799,3 +799,25 @@ def test_connect_token_network(
     assert channels[0]['balance'] == expected_balance
     assert channels[1]['balance'] == expected_balance
     assert channels[2]['balance'] == expected_balance
+    assert channels[0]['state'] == 'opened'
```
Comment (konradkonrad) [inline raiden/tests/api/test_api.py]: you should use the constants like `raiden.transfer.state.CHANNEL_STATE_OPENED` to compare the state

File: docs/Rest-Api.rst [hunk 2]
```diff
@@ -481,6 +481,80 @@ Possible Responses
...
+
+Example Request
+^^^^^^^^^^^^^^^
+
+``PUT /api/v1/connection/0x2a65aca4d5fc5b5c859090a6c34d164135398226``
+
+with payload::
+
+    {
+        "funds": 1337,
```
Comment (konradkonrad) [inline docs/Rest-Api.rst]: this is invalid json as it stands, json does not allow for dangling commas

File: raiden/tests/utils/apitestcontext.py [hunk 3]
```diff
@@ -264,3 +264,37 @@ def expect_token_swap(self, **kwargs):
...
+            wait_for_settle=True,
+            timeout=30):
+
+        channels = self.get_all_channels_for_token(token_address)
+        for channel in channels:
+            channel.external_state.netting_channel.state = CHANNEL_STATE_SETTLED
+            channel.external_state._settled_block = 1
+
+    def get_all_channels_for_token(self, token_address):
+        channels = []
```
Comment (LefterisJP) [inline raiden/tests/utils/apitestcontext.py]: These 4 lines could easily be replaced by a list comprehension.
```python
channels = [channel if channel.token_address == token_address for channel in self.channels]
```

File: raiden/tests/fixtures/api.py [hunk 4]
```diff
@@ -70,19 +70,81 @@ def api_raiden_service(
...
-    monkeypatch.setattr(api, 'close', api_test_context.close)
-    monkeypatch.setattr(api, 'settle', api_test_context.settle)
-    monkeypatch.setattr(api, 'get_channel', api_test_context.get_channel)
-    monkeypatch.setattr(api, 'get_network_events', api_test_context.get_network_events)
-    monkeypatch.setattr(api, 'get_token_network_events', api_test_context.get_token_network_events)
-    monkeypatch.setattr(api, 'get_channel_events', api_test_context.get_channel_events)
-    monkeypatch.setattr(api, 'transfer', api_test_context.transfer)
-    monkeypatch.setattr(api, 'token_swap', api_test_context.token_swap)
-    monkeypatch.setattr(api, 'expect_token_swap', api_test_context.expect_token_swap)
+    monkeypatch.setattr(
```
Comment (czepluch) [inline raiden/tests/fixtures/api.py]: can do that yes. They were like they are now before @hackaugusto put them in one liners here https://github.com/czepluch/raiden/commit/deff21e7d2e6fc73d42b8ad1c9c94e815e02226c and this was done after I started working on this issue, so I just used them as they were before.

File: raiden/api/rest.py [hunk 5]
```diff
@@ -159,6 +166,8 @@ def __init__(self, raiden_api):
         self.tokens_list_schema = TokensListSchema()
         self.partner_per_token_list_schema = PartnersPerTokenListSchema()
         self.transfer_schema = TransferSchema()
+        self.connection_connect_schema = ConnectionConnectSchema()
+        self.connection_leave_schema = ConnectionsLeaveSchema()
```
Comment (czepluch) [inline raiden/api/rest.py]: Actually I just realized that I am not using these schemes anywhere anymore, so I might as well just remove this.

File: raiden/tests/utils/apitestcontext.py (hunk 3 above)
Comment (czepluch) [inline raiden/tests/utils/apitestcontext.py]: I think you mean
```python
channels = [channel for channel in self.channels if channel.token_address == token_address
```
right?

Comment (LefterisJP) [inline raiden/tests/utils/apitestcontext.py]: yeah :)
Should have checked it in an interpreter before posting it.

File: raiden/api/rest.py (hunk 5 above)
Comment (LefterisJP) [inline raiden/api/rest.py]: oh even better. Then remove the classes too.

File: raiden/api/rest.py [hunk 6]
```diff
@@ -199,6 +208,23 @@ def close(self, token_address, partner_address):
         result = self.channel_schema.dumps(channel_to_api_dict(raiden_service_result))
         return result
 
+    def connect(self, token_address, funds, initial_channel_target=3, joinable_funds_target=.4):
```
Comment (czepluch) [inline raiden/api/rest.py]: where would be the best/correct place to hard code these?

Comment (LefterisJP) [inline raiden/api/rest.py]: I am actually not sure. Only place I can think of is settings.py.

Comment (LefterisJP) [inline raiden/api/rest.py]: or constants.py

Comment (czepluch) [inline raiden/api/rest.py]: And also, do we want `wait_for_settle` to be True or False by default?

Comment (LefterisJP) [inline raiden/api/rest.py]: True

File: raiden/api/rest.py (hunk 5 above)
Comment (czepluch) [inline raiden/api/rest.py]: The classes are still used in `resources.py`

Comment (LefterisJP) [inline raiden/api/rest.py]: ah okay

**Pull request merged**